bytecode.py

```diff
@@ -345,11 +349,28 @@ def run_call(self, env: EvalEnv) -> Generator[EvalEnv, None, None]:
         func_env._local_env = {
             name: env[arg] for name, arg in zip(func_code.params, self.args)
         }
-        env[self.dest] = yield from func_code.run(func_env)
+        if self.specialization:
```

Comment:
Some test cases for this logic would be nice : )

Reply:
fiiineee :P